### PR TITLE
doc: dts: update howtos with new API

### PR DESCRIPTION
**Pull request by pabigot**
Document the use of DEVICE_DT_GET() to fetch device pointers at
compile time, and update the documentation on defining device
instances to use the devicetree macros.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

### CODE REVIEW

File: doc/guides/dts/howtos.rst [hunk 1]
```diff
@@ -508,16 +526,16 @@ devicetree to operate on specific device nodes:
...
-devicetree node data.
+   	DEVICE_DT_DEFINE(MYDEV(idx),					\
+   			my_dev_init_function,				\
+			device_pm_control_nop,				\
+			&my_data_##idx,					\
+			&my_cfg_##idx,					\
+			MY_DEV_INIT_LEVEL, MY_DEV_INIT_PRIORITY,	\
+			&my_api_funcs)
+
+Notice the use of APIs like :c:func:`DT_INST_PROP` and
```
Comment (mbolivar-nordic) [inline doc/guides/dts/howtos.rst:537]:
```suggestion
Notice the use of APIs like :c:func:`DT_PROP` and
```

**Review by mbolivar-nordic — COMMENTED** [review]
Just one thing. Thanks for this.

File: doc/guides/dts/howtos.rst (hunk 1 above)
Comment (pabigot) [reply doc/guides/dts/howtos.rst]: Done; you're welcome.

**Review by mbolivar-nordic — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by galak** [comment]
@pabigot can you rebase and fix the merge conflicts now that #31108 got merged.

**Comment by pabigot** [comment]
> can you rebase and fix the merge conflicts now that #31108 got merged.

@galak done

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
@mbolivar-nordic any plans to introduce a helper for DT that does the DEVICE_DT_GET & device_is_ready together similar to device_get_binding?

**Comment by mbolivar-nordic** [comment]
> @mbolivar-nordic any plans to introduce a helper for DT that does the DEVICE_DT_GET & device_is_ready together similar to device_get_binding?

None from me. The existing APIs are good enough for what I need.

**Pull request merged**